program_intent_engagement/apps/api/v1/views.py

```diff
@@ -1 +1,89 @@
-# Create your views here.
+"""
+V1 API Views
+"""
+from itertools import groupby
+
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from program_intent_engagement.apps.api.serializers import ProgramIntentSerializer
+from program_intent_engagement.apps.core.models import ProgramIntent
+
+
+class MostRecentAndCertainIntentsView(APIView):
+    """
+    Retrieve a list of user intents
+    This endpoint returns a list of intents specified by the user.
+      Grouped by programs and sorted by most recent
+      and CERTAIN (if available, otherwise MAYBE) intent
+    Path: /api/[version]/intents/most-recent-and-certain
+    Returns:
+     * 200: OK, list of intents
+     * 404: User not found
+    """
+
+    authentication_classes = (JwtAuthentication,)
+    serializer_class = ProgramIntentSerializer
+
+    @classmethod
+    def grab_best_intents(cls, user_intents):
+        """Given a list of intents sorted by program_uuid
```

Comment:
This comment is no longer accurate, because `user_intents` is not sorted.

Reply:
good catch